sleap_nn/model_trainer.py

```diff
@@ -239,14 +289,27 @@ def __init__(self, config: OmegaConf):
             backbone_config=self.model_config.backbone_config,
             head_configs=[self.model_config.head_configs],
         ).to(self.m_device)
-
-        # Initializing the model weights
-        if self.model_config.init_weights == "xavier":
-            self.model.apply(xavier_init_weights)
         self.training_loss = {}
         self.val_loss = {}
         self.learning_rate = {}
 
+        if self.model_config.init_weights == "xavier":
+            self.model.apply(xavier_init_weights)
+
+        if self.model_config.pre_trained_weights:
+            self._init_pretrain_model()
+
+    def _init_pretrain_model(self):
+        if self.model_config.backbone_config.backbone_type in ["swint", "convnext"]:
+            ckpt = eval(self.model_config.pre_trained_weights).DEFAULT.get_state_dict(
+                progress=True, check_hash=True
+            )
+            if self.model_config.backbone_config.backbone_config.in_channels == 1:
+                ckpt["features.0.0.weight"] = torch.unsqueeze(
+                    ckpt["features.0.0.weight"].mean(dim=1), dim=1
+                )
```

Comment:
Interesting decision... I'm not totally opposed to this, but have never seen it before so it makes me a little antsy.

Normally we'd either:
1. Just have a 1x1 conv with a linear projection to get the channels to match up, or
2. Replicate the single channel input image 3x along the channels axis so it's "RGB".

The second is what we do in core SLEAP and would probably be my preference. I'm willing to be convinced otherwise, but I'd prefer to not invent something new for such a fundamental early step of the model processing.